File: programmers/캐릭터의 좌표.py
def solution(keyinput, board):
    max_w = board[0] // 2
    max_h = board[1] // 2
    x, y = 0, 0
    direction = {"left": (x - 1, y), "right": (x + 1, y), "up": (x, y + 1), "down": (x, y - 1)}
    for k in keyinput:
        tx, ty = direction.get(k)
        x += tx
        y += ty
        if abs(x) > max_w or abs(y) > max_h:
            x -= tx
            y -= ty
    return [x, y]


print(solution(["left", "right", "up", "right", "right"], [11, 11]))
print(solution(["down", "down", "down", "down", "down"], [7, 9]))

File: programmers/코딩테스트 입문 with Java/캐릭터의좌표.java
package 코딩테스트입문;


/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120861
 */
public class 캐릭터의좌표 {
    public static void main(String[] args) {
        System.out.println(solution(new String[]{"left", "right", "up", "right", "right"}, new int[]{11, 11}));
        System.out.println(solution(new String[]{"down", "down", "down", "down", "down"}, new int[]{7, 9}));
        System.out.println(solution(new String[]{"down", "down", "down", "down", "down"}, new int[]{1, 1}));
        System.out.println(solution(new String[]{}, new int[]{7, 9}));
    }

    public static int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        int max_width = board[0] / 2;
        int max_height = board[1] / 2;
        for (String key : keyinput) {
            switch (key) {
                case "left":
                    if (answer[0] == -max_width) break;
                    answer[0]--;
                    break;
                case "right":
                    if (answer[0] == max_width) break;
                    answer[0]++;
                    break;
                case "up":
                    if (answer[1] == max_height) break;
                    answer[1]++;
                    break;
                case "down":
                    if (answer[1] == -max_height) break;
                    answer[1]--;
            }
        }
        return answer;
    }
}
